### DISCUSSION TITLE
Ubuntu 20.04 - Cutelyst install success, but no run in command line

### DISCUSSION THREAD

**Discussion by ricardopesce** [cutelyst/cutelyst#317] [Q&A] [answered]
Hello, I'm new to Linux (loving every discovery!!) and I'm returning to the programming field, after 10 years! C++ fan! 

I had some trouble finding and installing dependencies, but finally made it. And everytime I had to install another dependency, I removed the build directory and started all over again:

``` mkdir build && cd build ```

---------------------------------------------
Running cmake, the option that includes all dependencies didn't work:

"Or you can try to build all modules supported by your platform, this only works if the build directory is empty (first cmake run):"

``` cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_ALL=on ```

Here's the CMakeOutput.log for that:
[CMakeOutput.log](https://github.com/cutelyst/cutelyst/files/6757442/CMakeOutput.log)

So, removing and creating the build directory again, I ran this cmake command, and it worked, but lot's of plugins disabled:
```
cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DPLUGIN_VIEW_CLEARSILVER=on -DPLUGIN_VIEW_GRANTLEE=on -DPLUGIN_VIEW_EMAIL=on
```

> -DPLUGIN_VIEW_GRANTLEE=on -DPLUGIN_VIEW_EMAIL=on
> -- The CXX compiler identification is GNU 9.3.0
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Check for working CXX compiler: /usr/bin/c++ - skipped
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- CUTELYST_PLUGINS_DIR: /usr/lib/x86_64-linux-gnu/cutelyst3-qt5-plugins
> -- PLUGIN: View::Grantlee, enabled.
> -- PLUGIN: View::Cutelee, disabled.
> -- PLUGIN: View::ClearSilver, enabled.
> -- PLUGIN: View::Email, enabled.
> -- PLUGIN: Validator, enable PwQuality, disabled.
> -- PLUGIN: Memcached, disabled.
> -- PLUGIN: MemcachedSessionStore, disabled.
> -- PLUGIN: StaticCompressed, disabled.
> -- PLUGIN: CSRFProtection, disabled.
> -- PLUGIN: uWSGI, disabled.
> -- Found lrelease at /usr/bin/lrelease
> -- Translations are enabled
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /home/ricardo/progr/cutelyst/build

Here's the CMakeOutput.log file. 
[CMakeOutput.log](https://github.com/cutelyst/cutelyst/files/6757460/CMakeOutput.log)

----------------------------------------

The ' sudo make install ' completed successfully, but the package is not running in terminal:

```
~/progr/cutelyst/build$ cutelyst2 --version
cutelyst2: command not found

```
Here's the install_manifest.txt 
[install_manifest.txt](https://github.com/cutelyst/cutelyst/files/6757472/install_manifest.txt)

I don't know if it's just a matter of telling the SO (don't know how-to yet...), but there's no instructions about it in the cutelyst /wiki/Installing...

Hope someone can help me. Thanks in advance!

**Comment by dantti** [comment]
You compiled Cutelyst v3 it's another executable name

**Reply by ricardopesce** [reply]
Hello Daniel,

First of all, thank you very much for your promptness.

Ok: I found the names of the executables in the install_manifest.txt : 

191. /usr/bin/cutelystd3-qt5
192. /usr/bin/cutelyst3-qt5

~$ cutelyst3-qt5 --version
cutelyst 3.1.0
~$ cutelystd3-qt5 --version
cutelyst-wsgi 3.1.0

The thing is: I installed cutelyst3 just following the installation guide. 

Then, I headed for the Tutorial_02_CutelystBasics and adapted the exec call:
- ``` $ cutelyst3-qt5 --create-app Hello ``` (OK)

Running CMake, I got the Error:
> CMake Error at CMakeLists.txt:16 (find_package):
>   By not providing "FindCutelyst2Qt5.cmake" in CMAKE_MODULE_PATH this project
>   has asked CMake to find a package configuration file provided by
>   "Cutelyst2Qt5", but CMake did not find one.
> 
>   Could not find a package configuration file provided by "Cutelyst2Qt5" with
>   any of the following names:
> 
>     Cutelyst2Qt5Config.cmake
>     cutelyst2qt5-config.cmake
> 
>   Add the installation prefix of "Cutelyst2Qt5" to CMAKE_PREFIX_PATH or set
>   "Cutelyst2Qt5_DIR" to a directory containing one of the above files.  If
>   "Cutelyst2Qt5" provides a separate development package or SDK, be sure it
>   has been installed.

So, I went after cutelyst2 and found the link for the latest cutelyst2 release in your blog. 
 > https://github.com/cutelyst/cutelyst/releases/tag/v2.13.0

Then it got simple! I just installed the file cutelyst_2.13.0_amd64.deb and now, I'm following the tutorial smoothly, running cutelyst2!

PS: Maybe some explanation about the different versions in the tutorial would help others.

Thanks again and congratulations for cutelyst itself! :)

**Comment by dantti** [comment, marked as answer]
You don't need Cutelyst v2, it seems a bug that the create app created files for old version.

**Reply by ricardopesce** [reply]
Got it! 
Unfortunately, I don't know how to fix this bug 😞 
If you have any hints... I could try. Just remembering that I'm new to linux environment (and maybe that could be the main cause here 😆 )

Meanwhile, I'll be following the tutorial using the version (v2.13.0) that is running ok in my os.

Thanks again!
